File: 3174 clear-digits/solution.go
func clearDigits(s string) string {
    st := make([]rune, len(s))
    top := -1
    for _, c := range s {
        if top != -1 && unicode.IsDigit(c) {
            top-- 
        } else {
            top++
            st[top] = c
        }
    }
    return string(st[:top+1])
}

File: 3174 clear-digits/solution.java
class Solution {
    public String clearDigits(String s) {
        char[] st = new char[s.length()];
        int top = -1;
        for(char c:s.toCharArray()) {
            if(top!=-1 && Character.isDigit(c)) top--;
            else st[++top] = c;
        }
        return new String(st,0,top+1);
    }
}
